src/data-access/repositories/mongo/mongo-device-repository.ts

```diff
@@ -1,24 +1,56 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { BaseMongoRepository } from './base-mongo-repository'
 import { DeviceRepository } from '../interfaces/device-repository'
 import { Device } from '../../../model/entities/device'
 import { MongoDatabase } from './mongo-database'
-import { MongoDevice, MongoEntityConverter } from './mongo-entity-converter'
+import { INewsDevice } from '../../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../../model/entities/telemetrics-device'
 
-const DEVICE_COLLECTION_NAME: string = 'peripheralDevices'
+const DEVICE_COLLECTION_NAME: string = 'externalDevices'
 
 export class MongoDeviceRepository extends BaseMongoRepository implements DeviceRepository {
 
-  constructor(mongoDatabase: MongoDatabase, private readonly mongoEntityConverter: MongoEntityConverter) {
+  constructor(mongoDatabase: MongoDatabase) {
     super(mongoDatabase)
   }
 
   protected getCollectionName(): string {
     return DEVICE_COLLECTION_NAME
   }
 
-  public async getDevices(): Promise<Device[]> {
-    this.assertDatabaseConnection(MongoDeviceRepository.prototype.getDevices.name)
-    const mongoDevices: MongoDevice[] = await this.getCollection().find<MongoDevice>({}).toArray()
-    return this.mongoEntityConverter.convertToDevices(mongoDevices)
+  public async findAllDevices(): Promise<(Device | INewsDevice | TelemetricsDevice)[]> {
+    this.assertDatabaseConnection(MongoDeviceRepository.prototype.findAllDevices.name)
+    const devices: Device[] = await this.getCollection().find<(Device | INewsDevice | TelemetricsDevice)>({}).toArray() 
+    return devices
+  }
+
+  public async findById(_deviceId: string): Promise<Device | null> {
+    this.assertDatabaseConnection(MongoDeviceRepository.prototype.findById.name)
+    try {
+      const device = await this.getCollection().findOne<(Device | INewsDevice | TelemetricsDevice)>({ _id: _deviceId })
+      return device
+    } catch (error) {
+      console.error(`Error finding device by id ${_deviceId}:`, error)
+      return null
+    }
+  }
+
+  public async create(_device: Device | INewsDevice | TelemetricsDevice): Promise<void> {
+    try{
+      this.assertDatabaseConnection(MongoDeviceRepository.prototype.create.name)
+      // const result: InsertOneResult<unknown> = 
+      await this.getCollection().insertOne(_device)
+    } catch (error) {
+      console.error(`Error creating new device by id ${_device.id}:`, error)
+    }
```

Comment:
No need to wrap it in a try-catch.
Here is an example of how we have implemented a create function elsewhere:
![image](https://github.com/tv2/sofie-server/assets/23049647/6a1623cb-ce59-4ccb-a797-b4768f94d4b6)